include/triton/core/tritonbackend.h

```diff
@@ -1427,16 +1427,34 @@ TRITONSERVER_Error* TRITONBACKEND_ModelBatchIncludeRequest(
 /// \param model The backend model for which Triton is forming a batch.
 /// \param userp The placeholder for backend to store and retrieve information
 /// about this pending batch.
+/// \param cache_userp The read-only placeholder for backend to retrieve
+// information about the batching strategy for this model.
 /// \return a TRITONSERVER_Error indicating success or failure.
 TRITONSERVER_Error* TRITONBACKEND_ModelBatchInitialize(
-    TRITONBACKEND_Model* model, void** userp);
+    TRITONBACKEND_Model* model, void** userp, const void* cache_userp);
 
 /// Callback to be invoked when Triton has finishing forming a batch.
 /// \param userp The placeholder for backend to store and retrieve information
 /// about this pending batch.
 /// \return a TRITONSERVER_Error indicating success or failure.
 TRITONSERVER_Error* TRITONBACKEND_ModelBatchFinalize(void* userp);
 
+/// Callback to be invoked when Triton loads model.
+/// This will hold a cached user pointer that can be read during custom
+/// batching.
+/// \param model The backend model for which Triton is forming a batch.
+/// \param cache_userp The placeholder for backend to store and retrieve
+/// information about the batching strategy for this model.
+/// return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_Error* TRITONBACKEND_ModelBatchCacheInitialize(
+    TRITONBACKEND_Model* model, void** cache_userp);
+
+/// Callback to be invoked when Triton unloads model.
+/// \param cache_userp The placeholder for backend to store and retrieve
+/// information about the batching strategy for this model.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_Error* TRITONBACKEND_ModelBatchCacheFinalize(void* cache_userp);
```

Comment:
You should talk about the lifecycle of this object, "init" called after model (even instance?) init and "final" before model (instance?) final

Reply:
Clarified the comments to make this clearer. Init/finalize is called during model init and finialization.